command/watch.go

```diff
@@ -0,0 +1,147 @@
+		ArgsUsage: "<cmd>",
+		Flags: []cli.Flag{
+			cli.Float64Flag{Name: "interval,n", Usage: "watch interval (sec)", Value: 2},
+			cli.IntFlag{Name: "priority,p", Usage: "Set the priority"},
+			cli.StringFlag{Name: "exec,x", Usage: "Pass command to exec (default to \"sh -c\")", Value: "sh -c"},
+			cli.StringFlag{Name: "title,t", Usage: "Set the title (empty for command)"},
+			cli.StringFlag{Name: "token", Usage: "Override the app token"},
+			cli.StringFlag{Name: "url", Usage: "Override the Gotify URL"},
+			cli.StringFlag{Name: "output,o", Usage: "Output verbosity (short|default|long)", Value: "default"},
+		},
+		Action: doWatch,
+	}
+}
+
+func doWatch(ctx *cli.Context) {
+	conf, confErr := config.ReadConfig(config.GetLocations())
+
+	cmdArgs := ctx.Args()
+	cmdStringNotation := strings.Join(cmdArgs, " ")
+	execArgs := strings.Split(ctx.String("exec"), " ")
+	cmdArgs = append(execArgs[1:], cmdStringNotation)
+	execCmd := execArgs[0]
+
+	outputMode := ctx.String("output")
+	if !(outputMode == "default" || outputMode == "long" || outputMode == "short") {
+		utils.Exit1With("output mode should be short|default|long")
+		return
+	}
+	interval := ctx.Float64("interval")
+	priority := ctx.Int("priority")
+	title := ctx.String("title")
+	if title == "" {
+		title = cmdStringNotation
+	}
+	token := ctx.String("token")
+	if token == "" {
+		if confErr != nil {
+			utils.Exit1With("token is not configured, run 'gotify init'")
+			return
+		}
+		token = conf.Token
+	}
+	stringURL := ctx.String("url")
+	if stringURL == "" {
+		if confErr != nil {
+			utils.Exit1With("url is not configured, run 'gotify init'")
+			return
+		}
+		stringURL = conf.URL
+	}
+	parsedURL, err := url.Parse(stringURL)
+	if err != nil {
+		utils.Exit1With("invalid url", stringURL)
+		return
+	}
+
+	watchInterval := time.Duration(interval*1000) * time.Millisecond
+
+	evalCmdOutput := func() (string, error) {
+		cmd := exec.Command(execCmd, cmdArgs...)
+		timeOut := time.After(watchInterval)
+		outputBuf := bytes.NewBuffer([]byte{})
+		cmd.Stdout = outputBuf
+		cmd.Stderr = outputBuf
+		err := cmd.Start()
+		if err != nil {
+			return "", fmt.Errorf("command failed to invoke: %v", err)
+		}
+		done := make(chan error)
+		go func() {
+			err := cmd.Wait()
+			if err != nil {
+				done <- fmt.Errorf("command failed to invoke: %v", err)
+			}
+			done <- nil
+		}()
+		select {
+		case err := <-done:
+			return outputBuf.String(), err
+		case <-timeOut:
+			cmd.Process.Kill()
+			return outputBuf.String(), errors.New("command timed out")
+		}
+	}
+
+	lastOutput, err := evalCmdOutput()
+	if err != nil {
+		utils.Exit1With("first run failed", err)
+	}
+	for range time.NewTicker(watchInterval).C {
+		output, err := evalCmdOutput()
+		if err != nil {
+			output += fmt.Sprintf("\n!== <%v> ==!", err)
+		}
+		if output != lastOutput {
+			msgData := bytes.NewBuffer([]byte{})
+			fmt.Fprintf(msgData, "command output for \"%s\" changed:\n\n", cmdStringNotation)
```

Comment:
I'd prefer if that is omited in short mode.

Reply:
There might be multiple watch commands running at the same time, so I think this piece of information should be included

Reply:
If it is actually needed, then they can use another verbosity, or just add it to the message themself. 
In view of https://github.com/gotify/cli/pull/20#discussion_r277072635 the actual command is irrelevant, it can even be executed mutiple times for different domains, as the domain is already part of the message.

Reply:
Okay